src/main/java/com/github/osisoft/ocs_sample_library_preview/BaseClient.java

```diff
@@ -112,21 +124,34 @@ public BaseClient(String apiVersion, String tenantId, String clientId, String cl
      * @return
      */
     public HttpURLConnection getConnection(URL url, String method) {
+        return getHttpURLConnection(url, method, Collections.<String, String>emptyMap());
```

Comment:
Maybe instead of passing an emptyMap you should pass getHttpHeadersForRequest()? This would eliminate the extra logic in the getHttpURLConnection

Reply:
Your absolutely right, updated. Thanks!